diff --git a/node_modules/react-native-router-flux/src/Router.js b/node_modules/react-native-router-flux/src/Router.js
index 27c3402..39bed4e 100644
--- a/node_modules/react-native-router-flux/src/Router.js
+++ b/node_modules/react-native-router-flux/src/Router.js
@@ -1,5 +1,5 @@
 import React from 'react';
-import { ViewPropTypes, BackHandler, Linking } from 'react-native';
+import { BackHandler, Linking } from 'react-native';
 import PropTypes from 'prop-types';
 import navigationStore from './navigationStore';
 import pathParser from './pathParser';
@@ -20,7 +20,7 @@ class App extends React.Component {
   };
 
   componentDidMount() {
-    BackHandler.addEventListener('hardwareBackPress', this.props.backAndroidHandler || this.onBackPress);
+    // BackHandler.addEventListener('hardwareBackPress', this.props.backAndroidHandler || this.onBackPress);
 
     // If the app was "woken up" by an external route.
     Linking.getInitialURL().then(url => this.parseDeepURL(url));
@@ -30,7 +30,7 @@ class App extends React.Component {
 
   componentWillUnmount() {
     BackHandler.removeEventListener('hardwareBackPress', this.props.backAndroidHandler || this.onBackPress);
-    Linking.removeEventListener('url', this.handleDeepURL);
+    // Linking.removeEventListener('url', this.handleDeepURL);
   }
 
   onBackPress = () => navigationStore.pop();
@@ -127,7 +127,7 @@ Router.propTypes = {
   navigator: PropTypes.func,
   wrapBy: PropTypes.func,
   getSceneStyle: PropTypes.func,
-  sceneStyle: ViewPropTypes.style,
+  sceneStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
   createReducer: PropTypes.func,
   children: PropTypes.element,
   uriPrefix: PropTypes.string,
